tests/micro-benchmarks/RecordFieldAttributes.fst

```diff
@@ -0,0 +1,111 @@
+module RecordFieldAttributes
+
+module T = FStar.Tactics
+
+type description (d : string) = ()
+
+type r = 
+    {
+        #[@@ (description "This is a number") ] field1 : int;
+        #[@@ (description "This is a string") ] field2 : string;
+    }
+
```

Comment:
Sadly, there's one aspect of it that I neglected to notice earlier. These attributes on binders are unfortunately coupled with implicit arguments (i.e., the preceding `#`) makes both these fields implicits arguments to the record constructor.

That means that the code below fails to typecheck

```
@@expect_failure]
let test (x:int) (y:string) = { field1 = x; field2 = y }
```

And you instead have to write:

```
let test (x:int) (y:string) : r = Mkr #x #y
```

I think we should decouple attributes on binders with implicit arguments. i.e., you should be able to associate an attribute with a binder whether it is implicit or not. E.g., 

```
([@@ attr1] x : t) -> (#[@@attr2] y:t) -> ... 
```

The first being an explicit and the second an implicit arg.

@aseemr @R1kM  ... wdyt? This should be fine for Steel too, right?

Reply:
I think it should be fine for Steel indeed. So far, I don't think we needed the framing attribute on explicit arguments, but this would be good to have for the future

Reply:
Right, I completely missed this.

"I think we should decouple attributes on binders with implicit arguments." 
@nikswamy Do I understand correctly that 'Meta' aqual cases are currently assumed to be implicit? And I should extend the aqual with some new case which could be used for an explicit argument? Plus change the rule so that when HASH is missing, we go with the new case?

Reply:
Yes, that's right.